File: src/eqn/ast/EqnAstNodeFunction.java
package eqn.ast;

abstract public class EqnAstNodeFunction extends EqnAstNode {
    public EqnAstNodeFunction(PrecedenceType precedenceType) {
        super(precedenceType);
    }

    public EqnAstNodeFunction(String value, Type type, PrecedenceType precedenceType) {
        super(value, type, precedenceType);
    }

    protected int arity() {
        return operands.size();
    }

    @Override
    public String toString() {
        StringBuffer stringBuffer = new StringBuffer(value + '(');
        for (int i = 0; i < operands.size() - 1; ++i) {
            stringBuffer.append(operands.elementAt(i).toString() + ", ");
        }
        stringBuffer.append(operands.lastElement().toString());

        stringBuffer.append(")");
        return new String(stringBuffer);
    }
}

File: src/eqn/ast/EqnAstNodeFunction.kt
package eqn.ast

abstract class EqnAstNodeFunction : EqnAstNode {
    constructor(precedenceType: PrecedenceType) : super(precedenceType)
    constructor(value: String, type: Type, precedenceType: PrecedenceType) : super(value, type, precedenceType)

    override fun arity(): Int {
        return operands.size
    }

    override fun toString(): String {
        val stringBuffer = StringBuffer("$value(")
        for (i in 0 until operands.size - 1) {
            stringBuffer.append(operands.elementAt(i).toString() + ", ")
        }
        stringBuffer.append(operands.lastElement().toString())
        stringBuffer.append(")")
        return String(stringBuffer)
    }
}